### PR TITLE
[FIX] - Incoming SMSs no longer clash with ongoing livechat conversations by the same visitor

### PR DESCRIPTION
**Pull request by cauefcr**
## Proposed changes (including videos or screenshots)

There was a data race in the defineVisitor function, causing new guests to be created even if a registered guest with that number already existed, also made sure that the open room being searched on is the correct source type, so the clash is not possible anymore.

### CODE REVIEW

**Review by murtaza98 — DISMISSED** [review]
LGTM!

File: apps/meteor/app/models/server/models/LivechatRooms.js [hunk 1]
```diff
@@ -288,13 +288,14 @@ export class LivechatRooms extends Base {
...
 
-	findOneOpenByVisitorTokenAndDepartmentId(visitorToken, departmentId, options) {
+	findOneOpenByVisitorTokenAndDepartmentIdAndSource(visitorToken, departmentId, source, options) {
 		const query = {
 			't': 'l',
 			'open': true,
 			'v.token': visitorToken,
 			departmentId,
 		};
+		if (source) query['source.type'] = source;
```
Comment (KevLehman) [inline apps/meteor/app/models/server/models/LivechatRooms.js:298]: We don't use short ifs 🙈 (I like them, but it's on the styleguide haha)
```suggestion
		if (source) {
			query['source.type'] = source;
		}
```

**Review by KevLehman — CHANGES_REQUESTED** [review]

File: apps/meteor/app/models/server/models/LivechatRooms.js (hunk 1 above)
Comment (cauefcr) [reply apps/meteor/app/models/server/models/LivechatRooms.js]: fixed!

**Review by KevLehman — CHANGES_REQUESTED** [review]
Will a conversation coming from Widget or from an app collide with an already open room? 👀 if so, we can use this PR to change that behavior to 🤗

**Review by KevLehman — APPROVED** [review]
Better let's try that one outside here 👀

**Pull request merged**